qiskit_nature/problems/second_quantization/electronic/electronic_structure_problem.py

```diff
@@ -52,19 +53,25 @@ def __init__(
             transformers: A list of transformations to be applied to the driver result.
         """
         super().__init__(driver, transformers)
+        self.main_property_name = "ElectronicEnergy"
```

Comment:
Did you consider adding some main_property name to the base problem constructor. It can default to "" there, but allow derived classes to pass up a different one. These seems to be constant for a given problem type so I can see a getter in the base class to get the value. Having a setter is less than obvious to me - is there ever a reason to set it outside of derived class constructor?

Reply:
I actually already have this in `BaseProblem`. You probably didn't notice because this line is actually using the setter method. But you're right, that a setter should never be used outside of the constructor. I'm going to remove that setter method in favor of setting the private variable during construction directly :+1: